shared/js/dialer/keypad.js

```diff
@@ -312,6 +312,14 @@ var KeypadManager = {
     this._longPress = false;
     this._lastPressedKey = key;
 
+    // If user input number more 50 digits, app shouldn't accept.
+    // The limit only applies while not on a call - there is no
+    // limit while on a call (bug 917630).
+    if (key !== 'delete' && !this._onCall &&
+        (this._phoneNumber.length >= this.kMaxDigits)) {
+      return;
+    }
+
```

Comment:
Do we need to make these changes on emergency-call keypad.js?  I was just thinking there could be some emergency related implications there, but haven't thought through all of them.